src/estimation.jl

```diff
@@ -44,29 +44,19 @@ function esprit(x::AbstractArray, M::Integer, p::Integer, Fs::Real=1.0)
 end
 
 """
-    jacobsen(x::Vector, Fs::Real = 1.0)
+    jacobsen(x::AbstractVector, Fs::Real = 1.0)
 
-Estimate the largest frequency (in Hz) in the signal `x` using Jacobsen's
-algorithm [^Jacobsen2007].
+Estimate the largest frequency in the signal `x` using Jacobsen's algorithm
+[^Jacobsen2007]. Argument `Fs` is the sampling frequency.  All frequencies are
+expressed in Hz.
+
+If the sampling frequency `Fs` is not provided, then it is assumed that `Fs =
+1.0`.
 
 [^Jacobsen2007]: E Jacobsen and P Kootsookos, "Fast, Accurate Frequency Estimators", Chapter
 10 in "Streamlining Digital Signal Processing", edited by R. Lyons, 2007, IEEE Press.
 """
-function jacobsen(x::Vector{<:Real}, Fs::Real = 1.0)
-    N = length(x)
-    X = rfft(x)
-    k = argmax(abs.(X))  # index of DFT peak
-    if ((k != div(N,2)+1) && (k != 1))  # avoid out-of-bounds indexing
-        # Jacoben's formula
-        δ = -real((X[k+1] - X[k-1]) / (2X[k] - X[k-1] - X[k+1]) )
-    else
-        δ = 0.0
-    end
-
-    return (k - 1 + δ)*Fs/N
-end
-
-function jacobsen(x::Vector{<:Complex}, Fs::Real = 1.0)
+function jacobsen(x::AbstractVector, Fs::Real = 1.0)
     N = length(x)
     X = fftshift(fft(x))
     k = argmax(abs.(X))  # index of DFT peak
```

Comment:
For real `x`, it appears to be a matter of chance which of the two maxima (due to the symmetry) we get. Should we limit the search range to 0 ... Fs/2 in that case?

Reply:
Well, since the algorithm (as mentioned in my comment above) is designed for complex signals, if someone uses it with real signals, caveat emptor applies...

I'll be happy to implement this in the way that you prefer. I can add a warning to the docs, or I can force the return value to be positive, or even restrict the signal type to complex.

Reply:
I decided to make these changes:

* Add explanation/warning about using Jacobsen with real signals to Jacobsen's docstring.
* Also document that, if input is real, the frequency estimate will be positive.
* Add tests for Jacobsen with real signals. The signal's frequency  in these tests is close to Fs/4, where the estimation error is at its lowest.

Reply:
Sounds good.